app/controllers/facility_user_reservations_controller.rb

```diff
@@ -0,0 +1,52 @@
+class FacilityUserReservationsController < ApplicationController
+
+  admin_tab :all
+  before_action :init_current_facility
+  before_action :authenticate_user!
+  before_action :check_acting_as
+  before_action :load_user
+
+  load_and_authorize_resource class: "OrderDetail"
+
+  layout "two_column"
+
+  # GET /facilities/:facility_id/users/:user_id/reservations
+  def index
+    @order_details = user_order_details
+                     .purchased
+                     .by_ordered_at
+                     .paginate(page: params[:page])
+  end
+
+  # PUT /facilities/:facility_id/users/:user_id/reservations/:id/cancel
+  def cancel
+    order_detail = user_order_details.find(params[:id])
+    raise ActiveRecord::RecordNotFound unless order_detail.reservation.try(:can_cancel?)
+    order_detail.transaction do
```

Comment:
I see this is coming from `OrderDetailsController`, but is the transaction necessary? Seems like if it failed, then nothing should have hit the database.